src/app/calculator/calculator.component.ts

```diff
@@ -14,65 +15,160 @@ import { MOCK } from '../shared/mockdata';
-      legPixel: null,
+      coordinateWaist: null,
+      coordinateKnee: null,
+      coordinateFootOnGround: null,
       footPixel: null,
-      isCorner: null,
+      isShouldDraw: false,
     };
-    let isShouldDraw = false;
-
     pict.passValue = (value) => {
-      isShouldDraw = !Object.values(value).some((v) => ((v === 0) || (v === null)));
+      draftValues.isShouldDraw = false;
+      draftValues.isShouldDraw = !Object.values(value).some((v) => ((v === 0) || (v === null)));
       Object.assign(draftValues, value);
     };
-
     pict.draw = () => {
-      if (isShouldDraw) {
+      if (draftValues.isShouldDraw) {
         pict.stroke(20, 233, 36);
         pict.strokeWeight(3);
-        if (draftValues.isCorner) {
-          pict.line(draftValues.xSaddle, draftValues.ySaddle,
-            draftValues.xSaddle + draftValues.xMedianCorner,
-            draftValues.ySaddle + draftValues.yInMiddle);
-          pict.line(draftValues.xSaddle, draftValues.ySaddle + draftValues.heightSaddlePixel,
-            draftValues.xSaddle + draftValues.xMedianCorner,
-            draftValues.ySaddle + draftValues.yInMiddle);
-          pict.line(draftValues.xSaddle, draftValues.ySaddle + draftValues.heightSaddlePixel,
-            draftValues.xSaddle + draftValues.footPixel,
-            draftValues.ySaddle + draftValues.heightSaddlePixel);
-        } else {
-          pict.line(draftValues.xSaddle, draftValues.ySaddle,
-            draftValues.xSaddle, draftValues.ySaddle + draftValues.legPixel);
-          pict.line(draftValues.xSaddle, draftValues.ySaddle + draftValues.legPixel,
-            draftValues.xSaddle + draftValues.footPixel,
-            draftValues.ySaddle + draftValues.legPixel);
+        pict.line(draftValues.coordinateWaist.x, draftValues.coordinateWaist.y,
+          draftValues.coordinateKnee.x, draftValues.coordinateKnee.y);
+        pict.line(draftValues.coordinateKnee.x, draftValues.coordinateKnee.y,
+          draftValues.coordinateFootOnGround.x, draftValues.coordinateFootOnGround.y);
+        pict.line(draftValues.coordinateFootOnGround.x, draftValues.coordinateFootOnGround.y,
+          draftValues.coordinateFootOnGround.x + draftValues.footPixel,
+          draftValues.coordinateFootOnGround.y);
+      } else {
+        console.log('sketchLeg: check your data');
+      }
+    };
+  }
+
+  sketchTorsAndArm = (p: any) => {
+    const pict = p;
+    const draftValues = {
+      coordinateWaist: {
+        x: null,
+        y: null,
+      },
+      coordinatePalmCenter: {
+        x: null,
+        y: null,
+      },
+      coordinateShoulder: {
+        x: null,
+        y: null,
+      },
+    };
+    let isShouldDraw = false;
+    pict.passValue = (value) => {
+      isShouldDraw = false;
+      Object.assign(draftValues, value);
+      isShouldDraw = !Object.values(draftValues).some((v) => {
+        if (v === null) {
+          return true;
         }
+        if ((v.hasOwnProperty('x')) && isNaN(v.x)) {
```

Comment:
why do you check for hasOwnProperty. I suppose you don't use inheritance

Reply:
yes.  I don't use it.
 but how to check this value? it can be null or object like {x:NaN, y: NaN} or at best {x:number, y:number}

Reply:
1) You can check for nan. 
2) why do u expect that value may be NaN?